mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -677,26 +676,23 @@ private boolean hasChildIn(Mailbox parentMailbox, List<Mailbox> mailboxesWithPat
     }
 
     @Override
-    public List<MessageId> search(MultimailboxesSearchQuery expression, MailboxSession session, long limit) throws MailboxException {
-        ImmutableSet<MailboxId> wantedMailboxesId =
-            getInMailboxes(expression.getInMailboxes(), session)
-                .filter(id -> !expression.getNotInMailboxes().contains(id))
-                .collect(Guavate.toImmutableSet());
-
-        return index.search(session, wantedMailboxesId, expression.getSearchQuery(), limit);
+    public Flux<MessageId> search(MultimailboxesSearchQuery expression, MailboxSession session, long limit) throws MailboxException {
+        return getInMailboxes(expression.getInMailboxes(), session)
+            .filter(id -> !expression.getNotInMailboxes().contains(id))
+            .collect(Guavate.toImmutableSet())
```

Comment:
what about using windowing?

Reply:
Not relevant as we rely on ElasticSearch to filter mailboxes all at one

Reply:
StoreMailboxManager has nothing to do with ES.
If we expect to always have all MessageIds at once, we should declare a `Mono<List<MessageId>>`

Reply:
Well the usual pattern with a search index is "push all of your constraint", let it deal with ordering/sorting, and consume it's results.

Some windowing on the mailboxIds is a non sens, independently from ES

Reply:
ok